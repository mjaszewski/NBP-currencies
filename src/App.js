import React, {Component} from "react";
import "./styles/main.scss";

import { Header, Container, Tab } from "semantic-ui-react";
import TableGroup from "./components/TableGroup"
import CurrencyCardGroup from "./components/CurrencyCardGroup"

class App extends Component {
    getPanes = () => [
        {
            menuItem: "Currency list",
            render: () => <Tab.Pane attached={false}><TableGroup /></Tab.Pane>,
        },
        {
            menuItem: "Favourite currencies",
            render: () => <Tab.Pane attached={false}><CurrencyCardGroup /></Tab.Pane>,
        },

    ];


    render() {
        return (
            <Container className="container">
                <Header as="h2" textAlign="center" >NBP Currencies</Header>
                <Tab menu={{ secondary: true, pointing: true }} panes={this.getPanes()} />
            </Container>
        );
    }
}

export default App

