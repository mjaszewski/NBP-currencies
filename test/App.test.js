import App from '../src/App';
import {Button} from "semantic-ui-react";

function shallowSetup() {
    const wrapper = shallow(<App />);

    return {
        wrapper
    };
}

let wrapper;

beforeEach(() => {
    const setup = shallowSetup();
    wrapper = setup.wrapper;
});

describe('Render App', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

