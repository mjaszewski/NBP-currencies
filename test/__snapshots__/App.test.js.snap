// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App renders correctly 1`] = `
<Container
  className="container"
>
  <Header
    as="h2"
    textAlign="center"
  >
    NBP Currencies
  </Header>
  <Tab
    grid={
      Object {
        "paneWidth": 12,
        "tabWidth": 4,
      }
    }
    menu={
      Object {
        "pointing": true,
        "secondary": true,
      }
    }
    panes={
      Array [
        Object {
          "menuItem": "Currency list",
          "render": [Function],
        },
        Object {
          "menuItem": "Favourite currencies",
          "render": [Function],
        },
      ]
    }
    renderActiveOnly={true}
  />
</Container>
`;
